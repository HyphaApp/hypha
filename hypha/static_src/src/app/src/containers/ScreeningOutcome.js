import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import SidebarBlock from '@components/SidebarBlock'
import { getSubmissionOfID } from '@selectors/submissions'


export const ScreeningOutcome = ({ submission }) => {
    const outcome = submission && submission.screening;
    return <SidebarBlock title="Screening Outcome">
        {  outcome ? outcome : "Not yet screened"}
    </SidebarBlock>
}

ScreeningOutcome.propTypes = {
    submission: PropTypes.object,
}

const mapStateToProps = (state, ownProps) => ({
    submission: getSubmissionOfID(ownProps.submissionID)(state),
})

export default connect(mapStateToProps)(ScreeningOutcome)
