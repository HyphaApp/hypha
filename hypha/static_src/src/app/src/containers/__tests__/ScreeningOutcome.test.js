import React from "react";
import {mount} from "enzyme";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({adapter: new Adapter()});
import { ScreeningOutcome } from '../ScreeningOutcome'

describe("Test screening outcome Container", () => {
    it("Should render screening outcome", () => {
        const submission = {screening : <div>Screening text</div>}
        const wrapper = mount(
                <ScreeningOutcome 
                submission={submission}
                />
        );
        expect(wrapper.find('div').length).toEqual(3)
        expect(wrapper.find('.sidebar-block').find('div').first().text().includes("Screening text")).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });

    it("Should render screening outcome without outcome", () => {
        const submission = {screening : null}
        const wrapper = mount(
                <ScreeningOutcome 
                submission={submission}
                />
        );
        expect(wrapper.find('div').length).toEqual(2)
        expect(wrapper.find('.sidebar-block').text().includes("Not yet screened")).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });
});