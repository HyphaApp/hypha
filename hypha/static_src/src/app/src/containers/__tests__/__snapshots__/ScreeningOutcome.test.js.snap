// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test screening outcome Container Should render screening outcome 1`] = `
<ScreeningOutcome
  submission={
    Object {
      "screening": <div>
        Screening text
      </div>,
    }
  }
>
  <SidebarBlock
    title="Screening Outcome"
  >
    <div
      className="sidebar-block"
    >
      <h5>
        Screening Outcome
      </h5>
      <div>
        Screening text
      </div>
      <div
        className="partition"
      >
         
      </div>
    </div>
  </SidebarBlock>
</ScreeningOutcome>
`;

exports[`Test screening outcome Container Should render screening outcome without outcome 1`] = `
<ScreeningOutcome
  submission={
    Object {
      "screening": null,
    }
  }
>
  <SidebarBlock
    title="Screening Outcome"
  >
    <div
      className="sidebar-block"
    >
      <h5>
        Screening Outcome
      </h5>
      Not yet screened
      <div
        className="partition"
      >
         
      </div>
    </div>
  </SidebarBlock>
</ScreeningOutcome>
`;
